### PR TITLE
Automatically replace water in Object Selection when clicking water types

### PR DESCRIPTION
**Pull request by 733737**
this overrides the error when displaying a palette that isn't selected

### CODE REVIEW

File: src/openrct2/EditorObjectSelectionSession.cpp [hunk 1]
```diff
@@ -573,12 +573,15 @@ ResultWithMessage WindowEditorObjectSelectionSelectObject(
...
+    if (maxObjects <= _numSelectedObjectsForType[EnumValue(objectType)] && objectType != ObjectType::Water)
     {
         return ObjectSelectionError(isMasterObject, STR_OBJECT_SELECTION_ERR_TOO_MANY_OF_TYPE_SELECTED);
     }
-
+    if (objectType == ObjectType::Water)
+    {
+        // Replace old palette with newly selected palette immediately.
+        ReplaceSelectedWaterPalette(item);
+    }
```
Comment (IntelOrca) [inline src/openrct2/EditorObjectSelectionSession.cpp:584]: Moving this here does not change the logic.

File: src/openrct2/EditorObjectSelectionSession.cpp [hunk 2]
```diff
@@ -47,7 +47,7 @@ static void ReplaceSelectedWaterPalette(const ObjectRepositoryItem* item);
 
 /**
  * Master objects are objects that are not
- * optional / required dependants of an
+ * optional / required dependents of an
```
Comment (IntelOrca) [inline src/openrct2/EditorObjectSelectionSession.cpp:50]: Why did you change this?

File: src/openrct2/EditorObjectSelectionSession.cpp [hunk 3]
```diff
@@ -681,7 +679,9 @@ int32_t EditorRemoveUnusedObjects()
 
                 // These object types require exactly one object to be selected at all times.
                 // Removing that object can badly break the game state.
-                if (objectType == ObjectType::ParkEntrance || objectType == ObjectType::Water)
+                if (objectType == ObjectType::ParkEntrance)
+                    continue;
+                if (objectType == ObjectType::Water)
                     continue;
```
[the remark this answers is not in the document]
Comment (733737) [reply src/openrct2/EditorObjectSelectionSession.cpp:685]: I agree

File: src/openrct2/EditorObjectSelectionSession.cpp (hunk 1 above)
Comment (733737) [reply src/openrct2/EditorObjectSelectionSession.cpp:584]: this is not a move, rather a hack to bypass the too many objects of this type selected error

File: src/openrct2/EditorObjectSelectionSession.cpp (hunk 2 above)
Comment (karst) [reply src/openrct2/EditorObjectSelectionSession.cpp:50]: isn't that just a difference between British and American English?

Comment (IntelOrca) [reply src/openrct2/EditorObjectSelectionSession.cpp:50]: > isn't that just a difference between British and American English?

Not necessarily. *-ant* is for the noun, *-ent* is for the adjective.

File: src/openrct2/EditorObjectSelectionSession.cpp (hunk 1 above)
Comment (IntelOrca) [reply src/openrct2/EditorObjectSelectionSession.cpp:584]: Surely that is resolved from the other change you did, not this particular one.

File: src/openrct2/EditorObjectSelectionSession.cpp (hunk 2 above)
Comment (733737) [reply src/openrct2/EditorObjectSelectionSession.cpp:50]: visual studio says dependants is not a word. But I changed it before visual studio even told me to.

**Comment by 733737** [comment]
this now works in a weird way. You open the water selection menu, uncheck whatever is selected, then just click on multiple palettes until you find the right one, then close the window. The palette you last clicked will be applied to your park, not the one with the checkmark. When you open the window again, you have to uncheck whatever palette is selected, then you can click as many palettes as you want before deciding one. It's a hacky improvement that people probably won't like. It works though, so I'm open to alternative ways of doing this.

**Comment by Manticore-007** [comment]
I had the same idea, I was talking to @Gymnasiast about it and he told me you are working on it. My idea was when you select a new palette/water stye, the previous one automatically gets deselected. Gymnasiast suggested the idea of a radio button for the palette, and that is actually what I would like to see.

**Comment by Broxzier** [comment]
I was actually the one who suggested that, but radio buttons do not exist yet. It should be easy enough to add them, though.

**Comment by Manticore-007** [comment]
> I was actually the one who suggested that

Oh, I was talking in DM to him about it 😁 but I think it would be a convenient thing for the palette choice 😎
